src/proctored-exam-settings/ProctoredExamSettings.jsx

```diff
@@ -0,0 +1,154 @@
+  );
+
+  return (
+    <div className="container">
+      <h2 className="mb-1">
+        Proctored Exam Settings
+      </h2>
+      <ValidationFormGroup
+        for="enableProctoredExams"
+        helpText="If checked, proctored exams are enabled in your course."
+      >
+        <CheckBox
+          id="enableProctoredExams"
+          name="enable_proctored_exams"
+          label="Enable Proctored Exams"
+          checked={enableProctoredExams}
+          onChange={onEnableProctoredExamsChange}
+        />
+      </ValidationFormGroup>
+      <ValidationFormGroup
+        for="allowingOptingOut"
+        helpText="If checked, learners can choose to take proctored exams without proctoring. If not checked, all learners must take the exam with proctoring."
+      >
+        <CheckBox
+          id="allowingOptingOut"
+          name="allow_opting_out"
+          label="Allow Opting Out of Proctored Exams"
+          checked={allowOptingOut}
+          onChange={onAllowOptingOutChange}
+        />
+      </ValidationFormGroup>
+      <ValidationFormGroup
+        for="proctoringProvider"
+        helpText="Select the proctoring provider you want to use for this course run."
+      >
+        <label htmlFor="proctoringProvider">Proctoring Provider</label>
+        <Input
+          type="select"
+          id="proctoringProvider"
+          name="proctoring_provider"
+          options={getProctoringProviderOptions(availableProctoringProviders)}
+          value={proctoringProvider}
+          onChange={onProctoringProviderChange}
+        />
+      </ValidationFormGroup>
+      <ValidationFormGroup
+        for="proctortrackEscalationEmail"
+        helpText="Required if 'proctortrack' is selected as your proctoring provider. Enter an email address to be contacted by the support team whenever there are escalations (e.g. appeals, delayed reviews, etc.)."
+      >
+        <label htmlFor="proctortrackEscalationEmail">Proctortrack Escalation Email</label>
+        <Input
+          type="text"
+          id="proctortrackEscalationEmail"
+          name="proctortrack_escalation_email"
+          value={proctortrackEscalationEmail}
+          onChange={onProctortrackEscalationEmailChange}
+        />
+      </ValidationFormGroup>
+      <ValidationFormGroup
+        for="createZendeskTickets"
+        helpText="If checked, a Zendesk ticket will be created for suspicious attempts."
+      >
+        <CheckBox
+          id="createZendeskTickets"
+          name="create_zendesk_tickets"
+          label="Create Zendesk Tickets For Suspicious Exam Attempts"
+          checked={createZendeskTickets}
+          onChange={onCreateZendeskTicketsChange}
+        />
+      </ValidationFormGroup>
+      <Button
+        className="btn-primary mb-3"
+        onClick={onButtonClick}
+      >
+        Submit
+      </Button>
+    </div>
+  );
+}
+
+ExamSettings.propTypes = {
+  id: PropTypes.string.isRequired,
+};
+
+ExamSettings.defaultProps = {};
```

Comment:
I think you need to specify a default prop for any required props. Do you see an error in the browser console about this?

Reply:
I wasn't seeing an error related to this, but I can definitely add one

Reply:
Just to update this thread, I tried adding a default prop, and the pre commit script said I should not have a default prop declared with a required propType

Reply:
Ack! I think I needed more coffee. It's the reverse - if you have a prop that _isn't_ required, you need to provide a default prop. Sorry about that!